recipe/thirdparty/providers/google.go

```diff
@@ -17,102 +17,179 @@ package providers
 
 import (
 	"encoding/json"
+	"errors"
 	"net/http"
 	"strings"
 
+	"github.com/derekstavis/go-qs"
 	"github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
 	"github.com/supertokens/supertokens-golang/supertokens"
 )
 
 const googleID = "google"
 
-func Google(config tpmodels.GoogleConfig) tpmodels.TypeProvider {
-	return tpmodels.TypeProvider{
-		ID: googleID,
-		Get: func(redirectURI, authCodeFromRequest *string, userContext supertokens.UserContext) tpmodels.TypeProviderGetResponse {
-			accessTokenAPIURL := "https://accounts.google.com/o/oauth2/token"
-			accessTokenAPIParams := map[string]string{
-				"client_id":     config.ClientID,
-				"client_secret": config.ClientSecret,
-				"grant_type":    "authorization_code",
-			}
-			if authCodeFromRequest != nil {
-				accessTokenAPIParams["code"] = *authCodeFromRequest
-			}
-			if redirectURI != nil {
-				accessTokenAPIParams["redirect_uri"] = *redirectURI
-			}
+func Google(input tpmodels.TypeGoogleInput) tpmodels.TypeProvider {
+	googleProvider := &tpmodels.GoogleProvider{}
 
-			authorisationRedirectURL := "https://accounts.google.com/o/oauth2/v2/auth"
-			scopes := []string{"https://www.googleapis.com/auth/userinfo.email"}
-			if config.Scope != nil {
-				scopes = config.Scope
-			}
+	getConfig := func(clientID *string, userContext supertokens.UserContext) (tpmodels.GoogleConfig, error) {
+		if input.Config == nil || len(input.Config) == 0 {
```

Comment:
this check should happen outside this function.

Reply:
Done

Reply:
GetConfig will throw error when there is no Config is provided, GetConfig is not overridden. This check will remain here.